force-app/main/default/classes/ALLO_Allocations_TEST.cls

```diff
@@ -1402,7 +1402,62 @@ private with sharing class ALLO_Allocations_TEST {
             insert newAllos;
             System.assert(false, 'Expected an exception due to Opportunity being Overallocated.');
         } catch (Exception e) {
-            System.assert(e.getMessage().contains(Label.alloTotalExceedsOppAmt), 'Expected Exception Text: ' + Label.alloExceedsOppAmount + '; Actual: ' + e.getMessage());
+            System.assert(e.getMessage().contains(Label.alloTotalExceedsOppAmt), 'Expected Exception Text: ' + Label.alloTotalExceedsOppAmt + '; Actual: ' + e.getMessage());
+        }
+
+        Test.stopTest();
+    }
+
+    /*******************************************************************************************************
+    * @description Tests that no error is generated if Opportunity Amount becomes 0 and a non zero fixed
+    * Amount Allocation exists
+    ********************************************************************************************************/
+    @isTest
+    private static void testFixupForNonZeroFixedAmountAllocationIfOpportunityAmountIsZero() {
+        List<General_Accounting_Unit__c> gaus = new List<General_Accounting_Unit__c>();
+
+        General_Accounting_Unit__c defaultGau = new General_Accounting_Unit__c(Name='General');
+        gaus.add(defaultGAU);
+        General_Accounting_Unit__c gau1 = new General_Accounting_Unit__c(Name = 'GAU 1');
+        gaus.add(gau1);
+
+        insert gaus;
+
+        setupSettings(new Allocations_Settings__c(Default_Allocations_Enabled__c = true, Default__c = defaultGau.Id));
+
+        List<Account> accs = UTIL_UnitTestData_TEST.createMultipleTestAccounts(1, null);
+
+        insert accs;
+
+        List<Opportunity> opps = UTIL_UnitTestData_TEST.oppsForAccountList(accs, null, UTIL_UnitTestData_TEST.getClosedWonStage(), System.today(), 10, null, null);
+
+        insert opps;
+
+        Test.startTest();
+
+        List<Allocation__c> newAllos = new List<Allocation__c>();
+
+        // Fix Amount Allocation for $10 to GAU 1
+        newAllos.add(
+            createOppAllocation(gau1.Id, opps[0].Id, 10, null)
+        );
+
+        insert newAllos;
+
+        try {
+            opps[0].Amount = 5;
+            update opps[0];
+            System.assert(false, 'Expected an exception due to Opportunity being Overallocated.');
+        } catch (Exception e) {
+            System.assert(e.getMessage().contains(Label.alloExceedsOppAmount), 'Expected Exception Text: ' + Label.alloExceedsOppAmount + '; Actual: ' + e.getMessage());
+        }
+
+        try {
+            opps[0].Amount = 0;
+            update opps[0];
+            System.assert(true, 'Expected no Exception since Opportunity had an Amount of 0');
```

Comment:
It will be clearer to have an assertion that checks the allocation amount are updated to 0 dollars.

Reply:
Great suggestion, just pushed the change.